src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
+            await TestMissingInRegularAndScriptAsync(code);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        [Trait("TODO", "Fix missing")]
+        public async Task TestInvocation_Missing_ExplicitInterfaceImplementation()
+        {
+            // error CS0539: 'C1.M1(int)' in explicit interface declaration is not a member of interface
+            var code =
+@"
+interface I1
+{
+    void M1();
+}
+class C1 : I1
+{
+        void I1.M1() { }
+        void I1.[|M1|](int i) { }
+}";
+            // Could apply argument to interface method: void M1(int i);
+            await TestMissingAsync(code);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_OverloadResolutionFailure()
+        {
+            // error CS1503: Argument 1: cannot convert from 'double' to 'int'
+            var code =
+@"
+    class C1
+    {
+        void M1(int i1, int i2) { }
+        void M1(double d) { }
+        void M2()
+        {
+            M1([|1.0|], 1);
+        }
+    }
+";
+            var fix0 =
+@"
+    class C1
+    {
+        void M1(int i1, int i2) { }
+        void M1(double d, int v) { }
+        void M2()
+        {
+            M1(1.0, 1);
+        }
+    }
+";
+            await TestInRegularAndScriptAsync(code, fix0);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_LambdaExpressionParameter()
+        {
+            // error CS1660: Cannot convert lambda expression to type 'int' because it is not a delegate type
+            var code =
+@"
+    class C1
+    {
+        void M1(int i1, int i2) { }
+        void M1(System.Action a) { }
+        void M2()
+        {
+            M1([|()=> { }|], 1);
+        }
+    }
+";
+            var fix =
+@"
+    class C1
+    {
+        void M1(int i1, int i2) { }
+        void M1(System.Action a, int v) { }
+        void M2()
+        {
+            M1(()=> { }, 1);
+        }
+    }
+";
+            await TestInRegularAndScriptAsync(code, fix);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_NamedParameter()
+        {
+            // error CS1739: The best overload for 'M1' does not have a parameter named 'i2'
+            var code =
+@"
+    class C1
+    {
+        void M1(int i1) { }
+        void M2()
+        {
+            M1([|i2|]: 1);
+        }
+    }
+";
+            var fix =
+@"
+    class C1
+    {
+        void M1(int i1, int i2) { }
+        void M2()
```

Comment:
what happen if M1(int, int) already exist but with different parameter name? like not_i2

Reply:
The fix offers to add a parameter of the given name to fix CS21739. It doesn't matter how much parameters are there already and what names they have (CS21739 is generated in the test even though the number and type of arguments and parameters match).

Reply:
I guess what you mean is it will add new parameter then? rather than new method? probably design question @kuhlenh @jinujoseph 

```csharp
void M1(int i1, int i3) { }
        void M1(int i1) { }

        void M2()
        {
            M1(i2: 1);
        }
```

I am not saying it can't be checked in until design issue is resolved. I am just letting pm knows in case she want to change design and open an issue.

Reply:
> I guess what you mean is it will add new parameter to one of overload?

Yes. The CodeFixer offers to add a parameter to either overload of M1.